src/System.IO.FileSystem/src/System/IO/UnixFileSystem.Linux.cs

```diff
@@ -0,0 +1,10 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace System.IO
+{
+    internal sealed partial class UnixFileSystem
+    {
+        public override bool CaseSensitive { get { return true; } }
```

Comment:
Wouldn't this depend on the actual mount point filesystem? There are many different file system formats for *nix systems and I'm sure someone somewhere formatted their system with a case insensitive file system. Should this do a best effort check for the root mount point and check if that is case sensitive?